website/content/docs/job-specification/change_script.mdx

```diff
@@ -0,0 +1,50 @@
+---
+layout: docs
+page_title: change_script Stanza - Job Specification
+description: The "change_script" stanza configures a script to be run on template re-render.
+---
+
+# `change_script` Stanza
+
+<Placement groups={['job', 'group', 'task', 'template', 'change_script']} />
+
+The `change_script` stanza allows operators to configure scripts that
+will be executed on template change. This stanza is only used when template
+`change_mode` is set to `script`. 
+
+```hcl
+job "docs" {
+  group "example" {
+    task "server" {
+      template {
+        source      = "local/redis.conf.tpl"
+        destination = "local/redis.conf"
+        change_mode = "script"
+        change_script {
+          path          = "/bin/foo"
+          args          = ["-verbose", "-debug"]
+          timeout       = "5s"
+          fail_on_error = false
+        }
+      }
+    }
+  }
+}
+```
+
+## `change_script` Parameters
+
+- `path` `(string: "")` - Specifies the full path to a script that
+  is to be executed on template change. Path is relative to the driver, e.g., if
+  running with a container driver the path must be existing in the container.
+  This option is required is the `change_mode` is `script`.
+
+- `args` `(array<string>: [])` - List of arguments that are passed to the script
+  that is to be executed on template change. 
+
+- `timeout` `(string: "5s")` - Timeout for script execution specified using a
+  label suffix like "30s" or "1h". 
+
+- `fail_on_error` `(bool: false)` - If `true`, Nomad will kill the task if the
+  script execution fails. If `false`, script failure will be logged but the task
+  will continue uninterrupted. 
```

Comment:
It would be good to have some examples here, there are some good ones in [`check`](https://www.nomadproject.io/docs/job-specification/check#script-checks-with-shells). One thing that I would add is one example using another `template` as the script:

```hcl
      template {
        data        = "{{key \"abc\"}}"
        destination = "local/test"
        change_mode = "script"

        change_script {
          path = "/local/script.sh"
        }
      }

      template {
        data        = <<EOF
#!/usr/bin/env bash
echo "Running change_mode script"
sleep 10
echo "Done"
EOF
        destination = "local/script.sh"
        perms       = "777"
      }
```

I forgot to set `perms` and add the shebang in early attempts 😅

Reply:
added your example 👍